web/hooks/use-long-touch.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect, useCallback } from 'react'
+
+/**
+ * Function to for dual functionality of button, on touch and on long touch (ms)
+ * This only works for touch, clicking will only do on click, as the intent of this function is to have a substitute for hover on touch devices
+ *
+ * KNOWN BUG: going from touch to clicking in the same session will require a refresh for the button to work, because touch overrides click
+ */
+export default function useLongTouch(
+  onLongTouch = () => {},
+  onClick = () => {},
+  ms = 500
+) {
+  const [startLongTouch, setStartLongTouch] = useState(false)
+  const [mouseState, setMouseState] = useState<
+    null | 'startPress' | 'endPress' | 'startTouch' | 'endTouch'
+  >(null)
+  const [timerId, setTimerId] = useState<
```

Comment:
I think this could be a `useRef` instead of `useState`, because you don't need to re-render. It's fine though.